packages/cqrs/lib/src/cqrs.dart

```diff
@@ -166,4 +311,47 @@ class Cqrs {
       },
     ).timeout(_timeout);
   }
+
+  void _log(
+    CqrsMethod method,
+    _ResultType result, [
+    Object? error,
+    StackTrace? stackTrace,
+    List<ValidationError> validationErrors = const [],
+  ]) {
+    if (_logger == null) {
+      return;
+    }
```

Comment:
Not exactly what I mean: since you are still being tormented with the nullable logger below. Pin to a local variable perhaps (`final logger = _logger`)?

Reply:
or we can wait for dart 3.2 where private fields are being promoted.